test/e2e-common.sh

```diff
@@ -102,6 +104,22 @@ function enable_monitoring(){
       --role roles/cloudtrace.agent
 }
 
+# The warm-up broker serves the following purposes:
+#
+# 1. When the broker data plane is created for the first time, it is expected
+# that there will be some delay before workload identity credential being fully
+# propagated. A warm-up broker will force the data plane to be created before
+# the real testing. This helps prevent the credential propagation delay causing
+# test flakiness.
+#
+# 2. The broker data plane will be GCed if there is no broker. Usually this would
+# happen before we dump all the pod logs in the cloud-run-events namespace. The
+# warm-up broker makes sure there is always one broker left and thus data plane pods
+# won't be deleted before we dump logs.
+function warmup_broker_setup(){
+  ko apply -f ${CONFIG_WARMUP_GCP_BROKER}
```

Comment:
Should we clean this up at the end of the tests?

Reply:
We don't reuse clusters. So I think it's fine to leak in-cluster resources.

Reply:
Yes, we do. Let me do that.

Reply:
@Harwayne Just confirmed with Chi. We can't reliably do resource cleanup with scripts in prow. What he did was to always delete all pubsub topics/subs before tests (https://github.com/google/knative-gcp/blob/master/test/e2e-secret-tests.sh#L85-L89).